src/AlgebraicGeometry/Schemes/Varieties/Types.jl

```diff
@@ -0,0 +1,94 @@
+@doc Markdown.doc"""
+    AbsAffineAlgebraicSet <: AbsSpec
```

Comment:
Per the naming conventions outlined in #1981 (not (yet) in the dev docs), `Abs` should always stand for "Absolute", and `Abstract` should not be abbreviated.

I don't mean this as a blocker for this PR, but we need to resolve this and either change this code (could be in a future PR), or adjust that rule.

CC @fieker

Reply:
We have tons of abstract types. The names will grow very long.

Reply:
So some abbreviation would be desirable for me. But we should rename this consistently in the schemes folder and not mix `Abstract` and `Abs` in the same ecosystem.

Reply:
> From my point of view, these types are for non-specialists. Algebraic geometers will know their way around schemes and not be scared by them. But beginners and users of algebraic geometry (as opposed to algebraic geometers) will need some more basic objects to use for their rather restricted purposes. As soon as they outgrow it, there are always schemes.
> 
> For these more restricted types, we should provide a optional argument which allows switching tests of the properties on and off. What the argument should default to, is then a question to be discussed.

In some cases I do not want to work with schemes myself and stay in safe world of varieties,
e.g. when I am considering function fields, rational equivalence and Weil divisors. 
Having the varieties **geometrically** integral assures that one can interpret the results more easily after a base change to an algebraically closed field. 
A question for @wdecker we have absolute primary decomposition over `QQ`. Therefore we can check if a variety is geometrically integral over `QQ`. Can we also do this over a finite field? 
Say I am integral but not geometrically integral. Is there a way to detect this (without computing the absolute primary decomposition)? In Oscar? For geometrically reduced this is no problem because (reduced + over a perfect field) => geometrically reduced.